### ISSUE TITLE
Еrrors in Tulog.ipynb notebook

### ISSUE THREAD

**Issue by Shanginre** [signals-dev/Orion#153]
* Orion version: 0.1.5
* Python version: 3.6.2 64 бит (AMD64)
* Operating System: Windows 64

### Description

1. the following code:
```
from utils import plot, plot_ts, plot_rws, plot_error, unroll_ts
```
is executed with an error:
_ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected._

2. the following code:
```
df = load_signal (signal)
```
is executed with an error:
_from pandas.core.dtypes.generic import (
             ImportError: cannot import name 'StringIO'_

**please correct errors or tell me how to fix them**

**Comment by sarahmish** [comment]
Hello @Shanginre 

Would you mind telling me which version of `pandas` you are using?

**Comment by Shanginre** [comment]
the thing is, on my local machine it works without error, but only on remote VM colab.research.google.com an error occurs.

yes, the reason is in pandas. Version in the Google virtual machine is 1.1.5. This is pandas 0.24.2 (this version works fine on my local machine) installation logs:

```! pip install pandas == 0.24.2```
_Installing collected packages: pandas
   Found existing installation: pandas 1.1.5
     Uninstalling pandas-1.1.5:
       Successfully uninstalled pandas-1.1.5
Successfully installed pandas-0.24.2_

![image](https://user-images.githubusercontent.com/11506233/104140057-41ac1900-53d1-11eb-8382-dabdb8833687.png)

can you execute the code on remote VM colab.research.google.com?

**Comment by sarahmish** [comment]
yes, when you uninstall the default versions in google colab, you have to restart the runtime and then you can run it without any issues.

To restart the runtime, click runtime -> restart runtime or shortcut `ctrl + M`, I'm showing a screenshot of the menu here. Then just rerun the cells as you would do normally.

<img width="1717" alt="photo" src="https://user-images.githubusercontent.com/40212131/104140841-827c4180-5381-11eb-97b1-45ba7f1726a5.png">

**Comment by hovikgas** [comment]
Even after downgrading Pandas in Google Colab, I still have the same errors due to the utils module:

```---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-d6cfd8e53f8c> in <module>()
      8 from sklearn.preprocessing import MinMaxScaler
      9 
---> 10 from utils import plot, plot_ts, plot_rws, plot_error, unroll_ts

ImportError: cannot import name 'plot' ```

**Comment by hovikgas** [comment]
Without downgrading Pandas, there is no utils module at all... must be deprecated? 

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-d6cfd8e53f8c> in <module>()
      8 from sklearn.preprocessing import MinMaxScaler
      9 
---> 10 from utils import plot, plot_ts, plot_rws, plot_error, unroll_ts

ModuleNotFoundError: No module named 'utils'
```

**Comment by sarahmish** [comment]
Hi @hovikgas!

After downgrading `pandas` and restarting the runtime, make sure you run the cell with
```
! git clone https://github.com/signals-dev/Orion.git
! mv Orion/notebooks/tulog/* .
```

`utils` is a file specific for the tulog to help with plotting the time series and visualizing the results, the above cells are necessary to download the file to the colab directory such that we can then import it.

You can check if the download was successful by looking at the _Files_ tab in colab, you should see `utils.py` there.

Let me know if this solves your issue

**Comment by hovikgas** [comment]
Thank you! Wow you are fast! By the time I edited my comment, you beat me to it!

**Comment by sarahmish** [comment]
Happy to help @hovikgas :)

I do agree, it would be much cleaner to define everything in the colab notebook but there are too many dependencies to include in one file which will be hard to follow..

Ideally, it would be great to have an easy import from Google drive, but that's difficult to do without authentication then drive mounting. Cloning the repo is the most straightforward approach to loading the necessary files at the moment.

**Comment by sarahmish** [comment]
Closing this issue since the question has been answered!

**Comment by cjtaylo-csu** [comment]
This issue and answer above was very helpful.

In case anyone else is trying to use Google Colab with TadGAN/Orion, I found the following steps (IN THIS ORDER) necessary to get TadGAN/Orion working like running on a local PC:

```
# Need to run, restart kernel, then re-run
!pip install pandas==0.24.2
!pip3 install orion-ml
! pip3 install 'urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1'
```
```
# Setup Google drive
import os
from google.colab import drive
drive.mount('/content/drive')

# Change to directory of this notebook
os.chdir("drive/MyDrive/$DIR")
```
```
# Run BEFORE connecting to GPU kernel
# Change TF to 1.15.2
%tensorflow_version 1.x

# Check that GPU is connected and is the correct version
import tensorflow as tf
print(tf. __version__) 
print(tf.test.is_built_with_cuda())
```
